src/main/java/tech/jhipster/lite/module/infrastructure/secondary/javadependency/gradle/GradleCommandHandler.java

```diff
@@ -188,38 +192,70 @@ private static GradleDependencyScope gradleDependencyScope(JavaDependency depend
 
   @Override
   public void handle(RemoveDirectJavaDependency command) {
-    versionsCatalog.retrieveDependencySlugsFrom(command.dependency()).forEach(this::removeDependencyFromBuildGradle);
-    versionsCatalog.removeLibrary(command.dependency());
+    versionsCatalog
+      .retrieveDependencySlugsFrom(command.dependency())
+      .stream()
+      .filter(dependencySlug -> dependencyExistsFrom(dependencySlug, command.buildProfile()))
+      .forEach(dependencySlug -> {
+        removeDependencyFromBuildGradle(dependencySlug, command.buildProfile());
+        versionsCatalog.removeLibrary(command.dependency());
+      });
+  }
+
+  @ExcludeFromGeneratedCodeCoverage(reason = "The exception handling is hard to test and an implementation detail")
+  private boolean dependencyExistsFrom(DependencySlug dependencySlug, Optional<BuildProfileId> buildProfile) {
```

Comment:
It would be nicer to use and read with the following signature:
```suggestion
private Predicate<DependencySlug> dependencyExistsFrom(Optional<BuildProfileId> buildProfile) {
    return dependencySlug -> {
      try {
        return dependencyLinePattern(dependencySlug, buildProfile).matcher(Files.readString(buildGradleFile(buildProfile))).find();
      } catch (IOException e) {
        throw GeneratorException.technicalError("Error reading build gradle file: " + e.getMessage(), e);
      }
    };
  }
```

That way you could use it as `.filter(dependencyExistsFrom(command.buildProfile()))`

Reply:
Thanks! I have to do this more often 👍